vendors/espressif/boards/ports/ble/nimble/iot_ble_hal_gatt_server.c

```diff
@@ -734,18 +752,40 @@ BTStatus_t prvBTSendResponse( uint16_t usConnId,
     struct ble_gatt_access_ctxt * ctxt = ( struct ble_gatt_access_ctxt * ) ulTransId;
 
     BTStatus_t xReturnStatus = eBTStatusSuccess;
+    uint8_t dst_buf[ BLE_MAX_ATTR_LEN ] = { 0 };
+    size_t mtu_value = ble_att_mtu(usConnId);
 
     if( prvValidGattRequest() )
     {
         if( ctxt && ( ( ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR ) || ( ctxt->op == BLE_GATT_ACCESS_OP_READ_DSC ) ) )
         {
+            ESP_LOGD(TAG, "Send response callback; long_read_flag = %d, xLen received = %d, offset = %d", long_read_flag, pxResponse->xAttrValue.xLen, virtual_offset);
+            if ( pxResponse->xAttrValue.xLen >= ( mtu_value - 1 ) )
+            {
+                /* Long read or blob request if data received is of ( MTU - 1 ) size and more */
+                long_read_flag = 1;
```

Comment:
`long_read_flag` is of type `bool`, so shouldn't we do `long_read_flag = true;` here?